l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -477,4 +490,61 @@ contract ExecutorFacet is Base, IExecutor {
     function _setBit(uint256 _bitMap, uint8 _index) internal pure returns (uint256) {
         return _bitMap | (1 << _index);
     }
+
+    /// @notice Calls the point evaluation precompile and verifies the output
+    // Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof.
+    // Also verify that the provided commitment matches the provided versioned_hash.
+    function _pointEvaluationPrecompile(bytes32 _versionedHash, bytes32 _inputPoint, bytes calldata _valueCommitmentProof) internal returns (bytes32) {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _inputPoint, _valueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.call(precompileInput);
+
+        // ToDo: Check output against spec, should have field elements prepended but other clients dont use it.
+        require(success, "failed to call point evaluation precompile");
+        uint256 result = abi.decode(data, (uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    function _verifyBlobInformation(bytes calldata _pubdataCommitments, bytes32 _blob1Hash, bytes32 _blob2Hash) internal returns (bytes32 blobCommitmentRollingHash) {
+        uint256 versionedHashIndex = 0;
+
+        // ToDo: This should be dynamic instead of being hardcoded to 2 blobs
+        require(_pubdataCommitments.length == PUBDATA_COMMITMENT_SIZE * 2, "bs");
+
+        for (uint256 i = 0; i < _pubdataCommitments.length; i += POINT_EVALUATION_INPUT_SIZE) {
+            (bool success, bytes memory data) = s.blobVersionedHashGetter.call(abi.encode(versionedHashIndex));
+            require(success, "vc");
+            bytes32 versionedHash = abi.decode(data, (bytes32));
+
+            // ToDo: this can be made to be dynamic where evalutationPoint = keccak256(abi.encode(versionedHash, hashes[i / PUBDATA_COMMITMENT_SIZE]));
+            bytes32 evalutationPoint;
+            if (i < PUBDATA_COMMITMENT_SIZE) {
+                evalutationPoint = keccak256(abi.encode(versionedHash, _blob1Hash));
```

Comment:
the correctness of the evaluation point's value will be validated in the circuits. There is no need to recalculate it (you can just provide it in the calldata).

Note that value of the point is validated, while the evaluation of the polynomial is not

Reply:
ok, do we not want to stop the operator from potentially picking a point that would either pass here and fail on proving or, picking a point that they know is going to evaluate to a specific value?

Reply:
yea i agree, given that its not as straight forward as just bit shifting either, lets have it come from the commitment calldata